lib/internal/errors.js

```diff
@@ -1122,7 +1129,26 @@ E('ERR_OUT_OF_RANGE',
     msg += ` It must be ${range}. Received ${received}`;
     return msg;
   }, RangeError);
-E('ERR_REQUIRE_ESM', 'Must use import to load ES Module: %s', Error);
+E('ERR_REQUIRE_ESM', (filename, module = null, pkg = null) => {
+  let msg = `Must use import to load ES Module: ${filename}\n`;
+  if (module && pkg) {
+    const parentPath = module.parent && module.parent.filename;
+    const basename = parentPath &&
+        lazyPath().basename(filename) === lazyPath().basename(parentPath) ?
+      filename : lazyPath().basename(filename);
+    msg +=
+      'require() of ES modules is not supported.\nrequire() of ' +
+      `${filename} ${parentPath ? `from ${module.parent.filename} ` : ''}` +
```

Comment:
@guybedford We need a 4 argument with `module.parent.filename` too, I'm ok with removing lazy path from error registering.

Reply:
Yes you're right.

Reply:
Actually no, `parentPath === module.parent.filename` :)

Reply:
sure? is not `parentPath = module.parent && module.parent.filename` a boolean?

Reply:
No, when using `&&` in JavaScript the value of the right hand side is returned by the expression if the left hand side is non-truthy.

Reply:
Thanks! I'll fix this, a very nice review!